src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java

```diff
@@ -1115,6 +1115,8 @@ public SolrInputDocuments toSolrDocs(IndexableDataset indexableDataset, Set<Long
             }
             LocalDate embargoEndDate=null;
             LocalDate end = null;
+            LocalDate retentionEndDate=null;
+            LocalDate start = null;
```

Comment:
A bug in the embargo code? Do start, and end (line 117) need to be defined inside the loop? Looks like they get set when the retention/embargo exist for a file but then aren't ever returned to null for the next file. It looks like that would cause files w/o retention/embargo to get their retention/embargo end dates set (~lines 1167-1172).

Reply:
Ok, if this is a bug then we need to fix it and all Dataverse instances with embargo enabled need to do a reindex.

Reply:
embargoEndDate and retentionEndDate must be outside the loop because those track the overall dates for the dataset; the latest embargo and the earliest retention expiration. 
But end and start should be inside the loop, as you suggest.